### PR TITLE
Cache the getnode/gettoken classifiers retrieived from the extension manager.

### PR DESCRIPTION
**Pull request by ToddGrun**
These showed up as ~6% of allocations in a find all references trace I'm looking at and it's trivial to cache these.

*** Previously ***
image

With my changes, these allocations are completely gone from the profile.

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs [hunk 1]
```diff
@@ -24,8 +25,17 @@ internal abstract class AbstractClassificationService(ISyntaxClassificationServi
 {
     private readonly ISyntaxClassificationService _syntaxClassificationService = syntaxClassificationService;
 
-    private static Func<SyntaxNode, ImmutableArray<Classifiers.ISyntaxClassifier>>? s_getNodeClassifiers;
-    private static Func<SyntaxToken, ImmutableArray<Classifiers.ISyntaxClassifier>>? s_getTokenClassifiers;
+    private static ExtensionClassifierInfo? s_cachedExtensionClassifierInfo;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: alternatively.  each classification service is lang specific.  so don't cache these statically.  cache in each particular subclass.

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: ooh, I like that better

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: Bummer, RemoteSemanticClassificationService accesses this method statically.

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: This seems like a common enough code path, that it should also hit the optimization, and I don't want to cache these both statically and on the instances, so I think I'd prefer to keep it as is unless you see a nice way around this.

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: you can have RemoteSemanticClassificationService fetch the real service, not access it statically.

Comment (ToddGrun) [reply src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs:28]: I think the general idea works (I don't think RemoteSemanticClassificationService  can access AbstractClassificationService, but I can just have the static method do the cast check.

**Pull request merged**